site/en/blog/fedcm-auto-reauthn-origin-trial/index.md

```diff
@@ -0,0 +1,175 @@
+
+<figure style="width: 300px; margin: auto; margin-top: 2em;">
+  {% Img
+    src="image/YLflGBAPWecgtKJLqCJHSzHqe2J2/jQvYTB6sHw5jGshyRxn6.png",
+    alt="A dialog the user taps on to create an account or to authenticate.", width="800", height="566", class="screenshot"
+  %}
+  <figcaption>A dialog the user taps on to create an account or to authenticate.</figcaption>
+</figure>
+
+The RP can request auto-reauthn by calling `navigator.credentials.get()` with `autoReauthn: true`.
+
+```js
+const cred = await navigator.credentials.get({
+  identity: {
+    providers: [{
+      configURL: "https://idp.example/fedcm.json",
+      clientId: "1234",
+    }],
+    // NOTE: We are exploring different API options to expose this
+    // functionality here:
+    // https://github.com/fedidcg/FedCM/issues/429#issuecomment-1426358523
+    // You should expect that, as a result of the origin trial, we'll
+    // learn more from developers and browser vendors what works best here.
+    autoReauthn: true, // default to false
+  },
+});
+
+```
+
+With this call, auto-reauthentication happens under the following conditions:
+
+* FedCM is available to use. For example, the user has not disabled FedCM either
+  globally or for the RP.
+* The user used only one FedCM account to sign into the website on this browser.
+* The user is signed into the IdP with that account.
+* The auto-reauthn didn't happen within the last 10 minutes.
+
+When the above conditions are met, an attempt to automatically reauthenticate the
+user starts as soon as the FedCM `navigator.credentials.get()` is invoked.
+
+
+<figure style="width: 300px; margin: auto; margin-top: 2em;">
+  {% Video
+    src="video/YLflGBAPWecgtKJLqCJHSzHqe2J2/eYnQxoDMNh4rpsyjcsQI.mp4",
+    autoplay="true", loop="true", class="screenshot"
+  %}
+  <figcaption>A user is auto-reauthenticating to an RP using FedCM</figcaption>
+</figure>
+
+{% Aside %}
+To avoid a frustrating experience of auto-reauthentication immediately after a
+user has signed out, it's recommended to design a sign-out flow that would prevent that.
+
+FedCM has a 10 minute quiet period after an auto-reauthentication to
+prevent this behavior. We are also exploring other approaches to achieve this
+such as using
+[`CredentialsContainer.preventSilentAccess()`](https://developer.mozilla.org/docs/Web/API/CredentialsContainer/preventSilentAccess).
+
+{% endAside %}
+
+## Try it out
+
+You can try FedCM auto-reauthentication locally by turning on [a Chrome
+flag](/docs/web-platform/chrome-flags/) `chrome://flags#fedcm-auto-re-authn` on
+Chrome 112 or later.
+
+For testing purposes, you can reset the 10 minute quiet period by removing
+browser data.
+
+1.  Navigate to `chrome://history`.
+2.  Click **Clear Browsing History**  (under the main menu).
+3.  Clear **Cookies & Other Site Data** for the time range **All time**.
+4.  Restart Chrome.
+
+{% Aside 'caution' %}
+
+Be careful that by doing this, you will lose cookies and storage data from all
```

Comment:
Sorry I didn't catch this earlier. Alternatively the developer can
1. Navigate to `chrome://history`.
2. in the search history box, search for the RP origin
3. click the 3 dot icon and remove it from history
4. Restart Chrome

This way the user won't lose any other cookies or browsing history on other sites.

Reply:
In that case I'll replace the guide entirely. That's easier and low friction.